File: ugly_number.cpp
#include "leetcode.h"

class Solution {
public:
  bool isUgly(int n) {
    if (n == 0)
      return false;
    for (int i : {2, 3, 5})
      while (n % i == 0)
        n /= i;
    return n == 1;
  }
};

int main() {
  Solution obj;
  if (obj.isUgly(6))
    cout << "true";
  else
    cout << "false";
}

File: other/ugly_number.rs
struct Solution;

impl Solution {
    pub fn is_ugly(mut n: i32) -> bool {
        if n == 0 {
            return false;
        }
        for i in vec![2,3,5] {
            while n % i == 0 {
                n /= i;
            }
        }
        n == 1
    }
}

fn main() {
    print!("{}", Solution::is_ugly(6));
}
